src/main/java/net/fabricmc/loom/configuration/ide/idea/IdeaSyncTask.java

```diff
@@ -110,6 +110,8 @@ private List<IntelijRunConfig> getRunConfigs() throws IOException {
 			irc.getExcludedLibraryPaths().set(excludedLibraryPaths);
 			irc.getLaunchFile().set(runConfigFile);
 			configs.add(irc);
+
+			settings.makeRunDir();
```

Comment:
Does this need applying to the other IDE sync tasks?

Also, it doesn't have to be in this PR but it would be good to support pre-run Gradle tasks. I know it's not ideal to run a gradle task before an IDE run but I think it is the least bad of all the ways to clean the run directory for client gametests.

Reply:
No, this was something I removed/broke in https://github.com/FabricMC/fabric-loom/pull/1166

I strongly oppose running Gradle before IDE runs, its not easy to in intelij and likely impossible in vscode/eclipse, and takes ages. We already clear the run dir before using one of the Gradle tasks, if your tests depend on that disable the IDE runs and solely rely on Gradle, it will be quicke, better supported and doesnt require any gross hacks to make it work.

Reply:
As dissussed we will do something about this, likely with a new entrypoint but I will do it as its own thing.